blog/2022-07-18-building-addon-introduction.md

```diff
@@ -0,0 +1,301 @@
+
+> Actually, this is how we install FluxCD before KubeVela v1.1
+
+1. Install FluxCD using Helm Charts or downloaded yaml manifests
+2. Manually download FluxCD-related Definition manifests and apply them
+
+Although it seems like only 2 steps is needed, we found it to be quite troublesome:
+
+1. Complicated installation: users are required to refer to documentations to manually install FluxCD and tackle possible errors
+2. Scattered resources: users need to obtain different resource files from different places
+3. Hard distribution: users having to download manifests manually makes it hard to distribute all these resources in a uniformed way
+4. No multi-cluster support: KubeVela emphasizes multi-cluster delivery a lot. However, manual installation makes multi-cluster support hard to carry out.
+
+KubeVela addons are born to solve these problems.
+
+## How KubeVela addons work
+
+**KubeVela addons are intrinsically OAM Applications** plus Definitions and other extensions. Since Definitions usually depend on other resources (e.g. certain operators), these dependencies are described as OAM Applications and stored in files called *Application description files*.
+
+Let's assume we want to build a Redis addon, which makes it possible to use Redis Components in Applications, to create a Redis cluster. Such addon will at least include a Redis Operator (to create Redis clusters) and a ComponentDefinition (to define what is a `redis` Component).
+
+The installation process of an addon includes installing Applications (which includes a Redis Operator), Definitions and etc.
+
+![](../docs/resources/addon-mechanism.jpg)
+
+## Create your own addon
+
+> Note: this guide only applies to KubeVela v1.5 and later
+
+We will take Redis addon as an example, and guide you through the whole process of making an addon. The full source code of this guide is located at [catalog/redis-operator](https://github.com/kubevela/catalog/tree/master/experimental/addons/redis-operator). As an introduction, we will not dive too deep into the details. If you are interested in official documentations, refer to [Make Your Own Addon](https://kubevela.io/docs/platform-engineers/addon/intro).
+
+**First, we will need to consider what our addon can do for users.** Let's say our Redis addon can provide a Component called `redis-failover`, which will create a whole Redis cluster with just one Component. **Then we will figure out how to achieve this.** To define a `redis-failover` Component, we write a ComponentDefinition. To be able to create a Redis Cluster, we use a [Redis Operator](https://github.com/spotahome/redis-operator).
+
+Now the goals are clear:
+
+- Write an OAM Application, which includes a Redis Operator. (refer to `template.cue` and `resources` directory in the full source code)
+- Write a ComponentDefinition, which defines Components called `redis-failover`. (refer to `definitions` directory in the full source code)
```

Comment:
please add a link to definition guide here https://kubevela.net/docs/platform-engineers/cue/basic plus some explanation